src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -2075,7 +2064,255 @@ let rectifyTermColumns () =
+    Excel.run(fun context ->
+        promise {
+            let assayWorkSheet = context.workbook.worksheets.getItem(identifier)
+            let range = assayWorkSheet.getUsedRange true
+            let _ =
+                assayWorkSheet.load(propertyNames = U2.Case2 (ResizeArray[|"name"|])) |> ignore
+                range.load(propertyNames = U2.Case2 (ResizeArray["values"]))
+
+            do! context.sync().``then``(fun _ -> ())
+
+            let values =
+                range.values
+                |> Seq.map (fun x ->
+                    x
+                    |> Seq.map (fun xx -> if xx.IsSome && not (String.IsNullOrEmpty(xx.Value.ToString())) then Some (xx.Value.ToString()) else None))
+
+            if Seq.length values > 1 then return Some (parseToMetadata values)
+            else return None
+        }
+    )
+
+/// <summary>
+/// Delete excel worksheet that contains top level metadata
+/// </summary>
+/// <param name="identifier"></param>
+let deleteTopLevelMetadata (identifier: string option) =
+    Excel.run(fun context ->
+        promise {
+            if identifier.IsNone then
+                return [InteropLogging.Msg.create InteropLogging.Error $"No identifier for the metadat top level sheet is available"]
+            else
+                let worksheets = context.workbook.worksheets
+
+                let _ = worksheets.load(propertyNames = U2.Case2 (ResizeArray[|"values"; "name"|]))
+
+                do! context.sync().``then``(fun _ -> ())
+
+                let metadataSheet = context.workbook.worksheets.getItem(identifier.Value)
+
+                let _ = metadataSheet.load(propertyNames = U2.Case2 (ResizeArray[|"name"|]))
+
+                metadataSheet.delete()
+
+                do! context.sync().``then``(fun _ -> ())
+
+                return [InteropLogging.Msg.create InteropLogging.Warning $"The work sheet {metadataSheet.name} has been deleted"]
+        }
+    )
+
+/// <summary>
+/// Converts a sequence of sequence of excel data into a resizearray, compatible with Excel.Range
+/// </summary>
+/// <param name="metadataValues"></param>
+let private convertToResizeArrays metadataValues =
+
+    let maxLength = metadataValues |> Seq.maxBy Seq.length |> Seq.length
+    metadataValues
+    |> Seq.map (fun row ->
+        let endRow = if Seq.length row < maxLength then Seq.append row (seq { None }) else row
+        endRow
+        |> Seq.map (fun column -> if column.IsSome then Some (column.Value :> obj) else None)
+        |> ResizeArray
+    )
+    |> ResizeArray
+
+open FsSpreadsheet
+
+/// <summary>
+/// Deletes the data contained in the selected worksheet and fills it afterwards with the given new data
+/// </summary>
+/// <param name="context"></param>
+/// <param name="fsWorkSheet"></param>
+/// <param name="seqOfSeqs"></param>
+let private updateWorkSheet (context:RequestContext) (fsWorkSheet:FsWorksheet) (seqOfSeqs:seq<seq<string option>>) =
```

Comment:
This function does not delete all data in a worksheet only all data in the range of the new data

Reply:
It should delete all data beause the first range accessed is:
let range = worksheet.getUsedRange true
This selects all cells that contain a value or are formatted and those are set to null.

In the next step, the cells are selected, that are in the range of the new data.